operators/multiclusterobservability/pkg/rendering/renderer_alertmanager.go

```diff
@@ -40,88 +40,105 @@ func (r *MCORenderer) newAlertManagerRenderer() {
 	}
 }
 
-func (r *MCORenderer) renderAlertManagerStatefulSet(res *resource.Resource,
-	namespace string, labels map[string]string) (*unstructured.Unstructured, error) {
+func (r *MCORenderer) renderAlertManagerStatefulSet(res *resource.Resource, namespace string, labels map[string]string) (*unstructured.Unstructured, error) {
 	u, err := r.renderer.RenderNamespace(res, namespace, labels)
 	if err != nil {
 		return nil, err
 	}
+
 	obj := util.GetK8sObj(u.GetKind())
 	err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, obj)
 	if err != nil {
 		return nil, err
 	}
+
 	crLabelKey := mcoconfig.GetCrLabelKey()
+	imagePullPolicy := mcoconfig.GetImagePullPolicy(r.cr.Spec)
 	dep := obj.(*v1.StatefulSet)
 	dep.ObjectMeta.Labels[crLabelKey] = r.cr.Name
+	dep.Name = mcoconfig.GetOperandName(mcoconfig.Alertmanager)
 	dep.Spec.Selector.MatchLabels[crLabelKey] = r.cr.Name
 	dep.Spec.Template.ObjectMeta.Labels[crLabelKey] = r.cr.Name
-	dep.Name = mcoconfig.GetOperandName(mcoconfig.Alertmanager)
 	dep.Spec.Replicas = mcoconfig.GetReplicas(mcoconfig.Alertmanager, r.cr.Spec.InstanceSize, r.cr.Spec.AdvancedConfig)
 
 	spec := &dep.Spec.Template.Spec
+	spec.NodeSelector = r.cr.Spec.NodeSelector
+	spec.Tolerations = r.cr.Spec.Tolerations
+	spec.ImagePullSecrets = []corev1.LocalObjectReference{{Name: mcoconfig.GetImagePullSecret(r.cr.Spec)}}
 
-	imagePullPolicy := mcoconfig.GetImagePullPolicy(r.cr.Spec)
-	spec.Containers[0].ImagePullPolicy = imagePullPolicy
-	args := spec.Containers[0].Args
+	if len(spec.Containers) != 4 {
+		return nil, fmt.Errorf("expected 4 containers in alertmanager statefulset, got %d", len(spec.Containers))
+	}
+
+	// set the container names for readability
+	alertManagerContainer := &spec.Containers[0]
+	configReloaderContainer := &spec.Containers[1]
+	oauthProxyContainer := &spec.Containers[2]
+	kubeRbacProxyContainer := &spec.Containers[3]
 
+	alertManagerContainer.ImagePullPolicy = imagePullPolicy
 	if *dep.Spec.Replicas > 1 {
 		for i := int32(0); i < *dep.Spec.Replicas; i++ {
-			args = append(args, "--cluster.peer="+
+			alertManagerContainer.Args = append(alertManagerContainer.Args, "--cluster.peer="+
```

Comment:
Can just use fmt.Sprint here instead of adding strings

Reply:
yep. I didn't add this btw, just moved things around. Needs a rebase now anyway so will take care of it